src/ITypefaceFinder.cpp

```diff
@@ -8,12 +17,59 @@ OsmAnd::ITypefaceFinder::~ITypefaceFinder()
 {
 }
 
-OsmAnd::ITypefaceFinder::Typeface::Typeface(const sk_sp<SkTypeface>& skTypeface_, hb_face_t* hbTypeface_)
-	: skTypeface(skTypeface_)
-	, hbTypeface(hbTypeface_, [](auto p) { hb_face_destroy(p); })
+OsmAnd::ITypefaceFinder::Typeface::Typeface(const sk_sp<SkTypeface>& skTypeface_,
+                                            TFontPtr hbFont_,
+                                            std::set<uint32_t> delCodePoints_,
+                                            uint32_t repCodePoint_)
+    : skTypeface(skTypeface_)
+    , hbFont(std::move(hbFont_))
+    , delCodePoints(delCodePoints_)
+    , repCodePoint(repCodePoint_)
 {
 }
 
 OsmAnd::ITypefaceFinder::Typeface::~Typeface()
 {
 }
+
+/*static*/
+ std::shared_ptr<OsmAnd::ITypefaceFinder::Typeface> OsmAnd::ITypefaceFinder::constructTypeface(
+            const sk_sp<SkTypeface>& skTypeface,
+            const std::shared_ptr<hb_blob_t>& hbBlob)
+{
+    if (nullptr == skTypeface || nullptr == hbBlob)
+    {
+        return nullptr;
+    }
+
+    const auto pHbTypeface = hb_face_create(hbBlob.get(), 0);
+    if (!pHbTypeface || pHbTypeface == hb_face_get_empty())
+        return nullptr;
+
+    //here calculating replacement symbols
+    std::set<uint32_t> delCodePoints;
+    uint32_t repCodePoint;
+    auto hb_font = TFontPtr(hb_font_create(pHbTypeface),
```

Comment:
Have you checked that hb_font is usable after hb_face was destroyed w/o issues?

Reply:
https://harfbuzz.github.io/harfbuzz-hb-ft.html#hb-ft-font-create

> it is the client program's responsibility to destroy `ft_face` *after* the `hb_font_t` font object has been destroyed.

Reply:
You are pointed onto different function `hb-ft-font-create`, instead of `hb-face-create`(https://harfbuzz.github.io/harfbuzz-hb-face.html#hb-face-create) - these a bit different functions.
However, I have already updated it.

Reply:
? I've pointed to `hb_font_create` as I;ve uteded to - since the line in question uses hb_font_create. What do you mean?

Reply:
If I got the docs right, if you create font from face, face needs to exist at least as long as font.

Reply:
You are pointed onto different functions `hb_ft_font_create() != hb_font_create()`

Reply:
Right, true, sorry. And in any case you've addressed the thing.